lib/manager/gomod/artifacts.ts

```diff
@@ -1,24 +1,40 @@
 import { ensureDir, outputFile, readFile } from 'fs-extra';
 import { join, dirname } from 'upath';
-import { exec } from '../../util/exec';
+import { exec, ExecOptions } from '../../util/exec';
 import { find } from '../../util/host-rules';
-import { getChildProcessEnv } from '../../util/exec/env';
 import { logger } from '../../logger';
 import { UpdateArtifactsConfig, UpdateArtifactsResult } from '../common';
 import { platform } from '../../platform';
 
+function getPreCommands(): string[] | null {
+  const credentials = find({
+    hostType: 'github',
+    url: 'https://api.github.com/',
+  });
+  let preCommands = null;
+  if (credentials && credentials.token) {
+    const token = global.appMode
+      ? `x-access-token:${credentials.token}`
+      : credentials.token;
+    preCommands = [
+      `git config --global url.\\\\"https://${token}@github.com/\\\\".insteadOf \\\\"https://github.com/\\\\"`,
+    ];
+  }
+  return preCommands;
+}
+
 export async function updateArtifacts(
   goModFileName: string,
   _updatedDeps: string[],
   newGoModContent: string,
   config: UpdateArtifactsConfig
 ): Promise<UpdateArtifactsResult[] | null> {
   logger.debug(`gomod.updateArtifacts(${goModFileName})`);
-  const customEnv = ['GOPATH', 'GOPROXY', 'GONOSUMDB'];
-  const env = getChildProcessEnv(customEnv);
-  env.GOPATH = env.GOPATH || join(config.cacheDir, './others/go');
-  await ensureDir(env.GOPATH);
-  logger.debug('Using GOPATH: ' + env.GOPATH);
+
+  const goPath = process.env.GOPATH || join(config.cacheDir, './others/go');
```

Comment:
Read-only usage of `process.env` is okay, don't think it should be abstracted away